src/main/scala/typechecker/Subtyping.scala

```diff
@@ -53,7 +53,7 @@ object Subtyping {
   }
 
   def isSubtype(sub: Type, sup: Type): Boolean = (sub, sup) match {
-    case (TSizedInt(v1), TSizedInt(v2)) => v1 <= v2
+    case (TSizedInt(v1, un1), TSizedInt(v2, un2)) => un1 == un2 && v1 <= v2
```

Comment:
Just for fun, I did a quick survey to see what other languages allow for conversions like this. C allows whatever, it seems, and both Go and Rust disallow both directions (i.e., they're incomparable).

Reply:
Well-timed Regehr tweet: https://twitter.com/johnregehr/status/1150771871748509696